finally is not passing observable through properly

**RxJS version:**
rxjs 5 beta 6

**Code to reproduce:**
```
  getSessions() {
    this.spinnerService.show();
    return <Observable<Session[]>>this.http
      .get(sessionsUrl)
      .map(res => this.extractData<Session[]>(res))
      .catch(this.exceptionService.catchBadResponse)
      .finally(() => this.spinnerService.hide());
  }
```

See code example here ...
https://github.com/johnpapa/event-view/commit/da21a922faf74aa3f2b6fbf1af5d1d666419b24e#diff-a115014ed5c151b197295d50f1ce2a54R24

**Expected behavior:**
TypeScript's `tsc` should think `getSessions` should return `Observable<Session[]>` 

**Actual behavior:**
TypeScript's `tsc` thinks `getSessions` is  returning `Observable<{}>` 

**Additional information:**
- When I comment out the `finally` TypeScript is happy
- When I cast the return type  as `<Observable<Session[]>>` tsc is happy

**Error message:**
```
> tsc -p ./src/client

src/client/app/+sessions/session-list/session-list.component.ts(34,9): error TS2322: Type '{}' is not assignable to type 'Session[]'.
```

cc // @david-driscoll @blesh @robwormald @wardbell

--- a/src/operator/finally.ts
+++ b/src/operator/finally.ts
@@ -16,7 +16,7 @@ export function _finally<T>(callback: () => void): Observable<T> {
 }
 
 export interface FinallySignature<T> {
-  <T>(callback: () => void): Observable<T>;
+  (callback: () => void): Observable<T>;
 }
 
 class FinallyOperator<T> implements Operator<T, T> {

